vms/evm/signer/kms_signer.go

```diff
@@ -109,22 +109,31 @@ func (s *KMSSigner) SignTx(tx *types.Transaction, evmChainID *big.Int) (*types.T
 	if err != nil {
 		return nil, err
 	}
-
-	sigBytes, err := s.getEthereumSignature(h, sigAsn1.R.Bytes, sigAsn1.S.Bytes)
+	sigBytes, err := s.recoverEIP155Signature(h, sigAsn1.R.Bytes, sigAsn1.S.Bytes)
 	if err != nil {
 		return nil, err
 	}
+
 	return tx.WithSignature(signer, sigBytes)
 }
 
 func (s *KMSSigner) Address() common.Address {
 	return s.eoa
 }
 
-// Recover the EIP-155 signature from the KMS signature
+// Recover the EIP-155 signature from the KMS signature.
 // KMS returns the signature in ASN.1 format, but the EIP-155 signature is in R || S || V format,
-// so we need to test both V = 0 and V = 1 against the recovered public key
-func (s *KMSSigner) getEthereumSignature(txHash []byte, rBytes []byte, sBytes []byte) ([]byte, error) {
+// so we need to test both V = 0 and V = 1 against the recovered public key.
+// Additionally, with EIP-2 S-values are capped at secp256k1n/2, so adjust that if necessary.
+func (s *KMSSigner) recoverEIP155Signature(txHash []byte, rBytes []byte, sBytes []byte) ([]byte, error) {
+	sBigInt := new(big.Int).SetBytes(sBytes)
```

Comment:
nit: what was our decision with using new(big.Int) vs big.NewInt()?

Reply:
My preference is to use the constructor. Changed now.